tests/resources/test_predictor.py

```diff
@@ -37,6 +37,29 @@ def test_build(valid_simple_ml_predictor_data, basic_predictor_report_data):
     assert predictor.description == 'Predicts z from input x and latent variable y'
 
 
+def test_delete():
+    pc = PredictorCollection(uuid.uuid4(), mock.Mock())
+    with pytest.raises(NotImplementedError):
+        pc.delete(uuid.uuid4())
+
+
+def test_archive(valid_label_fractions_predictor_data):
+    session = mock.Mock()
+    pc = PredictorCollection(uuid.uuid4(), session)
+    session.get_resource.return_value = valid_label_fractions_predictor_data
+
+    def _mock_put_resource(url, data):
+        """Assume that update returns the serialized predictor data."""
+        return data
+    session.put_resource.side_effect = _mock_put_resource
+    archived_predictor = pc.archive(uuid.uuid4())
+    assert archived_predictor.archived
+
+    session.get_resource.side_effect = NotFound("")
+    with pytest.raises(RuntimeError):
+        pc.archive(uuid.uuid4())
+
+
```

Comment:
Only testing the predictor b/c the AbstractModuleCollection methods will get exercised through that pathway?  Is there a reason not to just make that module a parent class to all resources?

Reply:
> Only testing the predictor b/c the AbstractModuleCollection methods will get exercised through that pathway?

Yes, and since we require 100% test coverage, we know it's being exercised thoroughly

> Is there a reason not to just make that module a parent class to all resources?

You mean make `AbstractModuleCollection` a parent to all resources? It's not relevant to all resource collections. This delete/archive behavior is particular to collections of modules